### ISSUE TITLE
SQLscript: "WAIT" command to avoid rollback error.

### ISSUE THREAD

**Issue by gramian** [ArcadeData/arcadedb#1699]
### ArcadeDB Version:

`24.6.1`

Using a transaction (`BEGIN` ... `COMMIT`) and a backup (`BACKUP DATABASE`) right afterwards in **SQLscript**,
frequently a "pending transaction" error is logged. Thus, I propose a new SQLscript command `WAIT`, which pauses until all pending transactions are completed.

### Expected behavior

SQLscript:
```sql
BEGIN;
something
COMMIT;
WAIT;
BACKUP DATABASE;
```

### Current behavior

SQLscript:
```sql
BEGIN;
something
COMMIT;
BACKUP DATABASE;
```

Log:
```
SEVER [BackupDatabaseStatement] Found pending transaction. Rolling it back before the backup...
```

**Comment by gramian** [comment]
Alternatively, this could be an argument to the `COMMIT` command, like:
```sql
COMMIT wait=true;
```

**Comment by lvca** [comment]
That issue is because the current database has an active transaction, so any changing is rollbacked:

```java
    if (context.getDatabase().isTransactionActive()) {
      LogManager.instance().log(this, Level.SEVERE, "Found pending transaction. Rolling it back before the backup...");
      context.getDatabase().rollback();
    }
```

It's not about concurrent transactions. Let's say you do this:

```sql
begin
insert into Test content {a:3}
backup database
```

That insert wound't be in the backup.

**Comment by gramian** [comment]
Exactly, but how do I know when all transactions are finished and I can backup without a rollback?

**Comment by lvca** [comment]
The rollback it's only for the current TX (it shouldn't be executed in a transaction the backup). About other concurrent TXs, the backup database waits to hold a lock as soon as they are all committed.

**Comment by gramian** [comment]
Got it, so you are saying a `WAIT` command is not needed, as the problem I am having is related to an uncommitted tx?

For context, this is happening in an async HTTP API command, and adding an additional `COMMIT` to the SQLscript reports an error saying no-active-tx. Could it be that the async SQLscript gets delayed notice of tx states then?

**Comment by gramian** [comment]
@lvca Could it be that the open TX is SQLscript itself as every SQLscript is a TX?

**Comment by lvca** [comment]
Possible. If the TX is just open, a rollback doesn't do anything.

**Comment by gramian** [comment]
Given `SQLscript` can now handle `BACKUP DATABASE` this can be closed.
